whittle/models/gpt/extract.py

```diff
@@ -8,9 +9,24 @@
 from whittle.models.gpt.blocks.mlp import GptNeoxMLP, LLaMAMLP
 from whittle.modules.layernorm import LayerNorm
 from whittle.modules.rmsnorm import RMSNorm
+from litgpt import Config
+
+
+def extract_sub_network(model: GPT, sub_network_config: Config) -> GPT:
+    """
+    Extracts a sub-network from a given model based on the specified sub-network configuration.
+    Copies relevant layers, weights, and configurations from the full model into a sub-network model.
+
+    Args:
+        model: The original, full GPT model from which the sub-network is extracted.
+        sub_network_config: Configuration object for the sub-network, containing the necessary
+                                     architecture specifications such as embedding size, number of heads,
+                                     and number of layers.
```

Comment:
The line breaks seem a little weird here, maybe follow what eddie does [here in amltk](https://github.com/automl/amltk/blob/5221d80fb4fec92f2d6f2fc73fce4eb16139c09f/src/amltk/pipeline/components.py#L169)